### PR TITLE
Add timeout parameter, validation and tests

### PR DESCRIPTION
**Pull request by RichardSmithONS**

### CODE REVIEW

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseError.scala [hunk 1]
```diff
@@ -221,4 +221,20 @@ object CountryBoostsInvalidError extends AddressResponseError(
...
+)
+
+object TimeoutTooSmallAddressResponseError extends AddressResponseError(
+  code = 44,
+  message = "Timeout parameter is too small, minimum (milliseconds) = 50"
+)
+
+object TimeoutTooLargeAddressResponseError extends AddressResponseError(
+  code = 45,
+  message = "Timeout parameter is too large, maximum (milliseconds) = *"
```
Comment (steve-thorne-ONS) [inline model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseError.scala:238]: Should there be a numerical value here?

Comment (RichardSmithONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseError.scala:238]: No. It substitutes the value from the config file.

**Review by steve-thorne-ONS — APPROVED** [review]
LGTM

File: model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseError.scala (hunk 1 above)
Comment (steve-thorne-ONS) [reply model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/address/AddressResponseError.scala:238]: Sweet.

**Pull request merged**